common/transport/transport-api/src/main/java/org/thingsboard/server/common/transport/service/DefaultTransportService.java

```diff
@@ -1047,11 +1048,9 @@ private void onDeviceUpdate(Device device) {
                         .setDeviceProfileIdLSB(deviceProfileIdLSB)
                         .setDeviceName(device.getName())
                         .setDeviceType(device.getType()).build();
-                if (device.getAdditionalInfo().has("gateway")
-                        && device.getAdditionalInfo().get("gateway").asBoolean()
-                        && device.getAdditionalInfo().has(OVERWRITE_ACTIVITY_TIME)
-                        && device.getAdditionalInfo().get(OVERWRITE_ACTIVITY_TIME).isBoolean()) {
-                    md.setOverwriteActivityTime(device.getAdditionalInfo().get(OVERWRITE_ACTIVITY_TIME).asBoolean());
+                JsonNode deviceAdditionalInfo = device.getAdditionalInfo();
```

Comment:
I saw the same condition in abstract gateway handler. Please, use the same constants instead od raw string `gateway`
```
if (deviceAdditionalInfo.has(GATEWAY_PROPERTY) && deviceAdditionalInfo.has(OVERWRITE_ACTIVITY_TIME)) {
            overwriteDevicesActivity = deviceAdditionalInfo.get(OVERWRITE_ACTIVITY_TIME).asBoolean();
        }
```